tests/reference/asr-test_dict3-d28f38f.stderr

```diff
@@ -1,5 +1,5 @@
-semantic error: Type mismatch in get's key value, the types must be compatible
+semantic error: Found type mismatch in 'get'.Type mismatch (found: 'i32', expected: 'str')
  --> tests/errors/test_dict3.py:5:15
   |
 5 |     x = y.get(1)
-  |               ^ type mismatch (found: 'i32', expected: 'str')
+  |               ^ 
```

Comment:
This seems like a change for the worse? We want to go the other direction and highlight `x` as well, and "attach" labels to both parts showing what the types are.

I would recommend not to change errors in this PR, which should only do refactoring and not change tests.

Reply:
I see --- the errors were possibly consolidated. That is fine to change some tests then, but I would still try to restore the previous error, and while doing it, even highlight (underline) the `x` as well.

Reply:
My bad --- in this error, we do not want to highlight `x`, as the mismatch is in the argument of the `get` method. So just one highlight, but I would still add some meaningful label, such as the one that was there before.

Reply:
Previously, it was because of a `Diagnostic` error. I'll pass that as the argument to restore the previous error.